olcUTIL_Geometry2D.h

```diff
@@ -1231,26 +1231,42 @@ namespace olc::utils::geom2d
 	template<typename T1, typename T2>
 	inline std::vector<olc::v_2d<T2>> intersects(const rect<T1>& r1, const rect<T2>& r2)
 	{
-		// TODO:
-		return {};
+		std::vector<olc::v_2d<T2>> ret;
+		for (size_t i = 0; i < 4; i++) {
+			for (size_t j = 0; j < 4; j++) {
+				auto v = intersects(r1.side(i), r2.side(j));
+				ret.insert(ret.end(), v.begin(), v.end());
+			}
+		}
+		return ret;
 	}
 
 	// intersects(c,r)
 	// Get intersection points where circle intersects with rectangle
 	template<typename T1, typename T2>
 	inline std::vector<olc::v_2d<T2>> intersects(const circle<T1>& c, const rect<T2>& r)
 	{
-		// TODO:
-		return {};
+		std::vector<olc::v_2d<T2>> ret;
+		for (size_t i = 0; i < 4; i++) {
+			auto v = intersects(c, r.side(i));
+			ret.insert(ret.end(), v.begin(), v.end());
+		}
+		return ret;
 	}

```

Comment:
It seems that line vs circle doesn't work properly.
https://imgur.com/a/b8hB9iy
https://imgur.com/a/4AUC6Ib

Reply:
In addition, this will always return at least 2 intersections on corners.
Probably not desirable.

Reply:
I see, this got real messy. I will go over them again. Thanks for your time, closing the pr.